### PR TITLE
Alerting: Add context.Context to RuleStore

### PR DESCRIPTION
**Pull request by grobinson-grafana**
**What this PR does / why we need it**:

This pull request adds `context.Context` to `RuleStore` as at present no queries in the rule store have a timeout or support cancelation.

### CODE REVIEW

**Review by yuri-tceretian — APPROVED** [review]
Nice! Together with https://github.com/grafana/grafana/pull/44939 it unblocks some nice changes.

File: pkg/services/ngalert/schedule/schedule_unit_test.go [hunk 1]
```diff
@@ -504,7 +504,7 @@ func TestSchedule_ruleRoutine(t *testing.T) {
 		// Now update the rule
 		newRule := *rule
 		newRule.Version++
-		ruleStore.putRule(&newRule)
+		ruleStore.putRule(context.TODO(), &newRule)
```
Comment (JohnnyQQQQ) [inline pkg/services/ngalert/schedule/schedule_unit_test.go:507]: Why do we use in the other tests `.Background()` and here `.TODO()`? Is there something special we need to think of in the future?

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM! Left one question.

File: pkg/services/ngalert/schedule/schedule_unit_test.go (hunk 1 above)
Comment (gerobinson) [reply pkg/services/ngalert/schedule/schedule_unit_test.go]: Thanks! I was meant to change these to `Background()`!

**Pull request merged**